zcash_client_sqlite/src/lib.rs

```diff
@@ -245,6 +245,32 @@ impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletRead for W
     type Error = SqliteClientError;
     type AccountId = AccountId;
 
+    fn validate_seed(
+        &self,
+        account_id: Self::AccountId,
+        seed: &SecretVec<u8>,
+    ) -> Result<bool, Self::Error> {
+        self.get_unified_full_viewing_keys().and_then(|keys| {
+            keys.get(&account_id).map_or(Ok(false), |ufvk| {
+                let usk = UnifiedSpendingKey::from_seed(
+                    &self.params,
+                    &seed.expose_secret()[..],
+                    account_id,
+                )
+                .map_err(|_| SqliteClientError::KeyDerivationError(account_id))?;
+
+                // Keys are not comparable with `Eq`, but addresses are, so we derive what should
+                // be equivalent addresses for each key and use those to check for key equality.
```

Comment:
This should work but is quite ugly. I don't see any reason why keys shouldn't implement `ConstantTimeEq`.

Reply:
We can do that; I was being a bit hasty trying to get this in so that the guys could use it in Zashi, and I didn't want to wait for `sapling-crypto` and `orchard` releases. I guess the change is purely additive, though, so we can do it in a point release.